toolkit/authentication/src/androidTest/java/com/arcgismaps/toolkit/authentication/OAuthDefaultConfigurationTests.kt

```diff
@@ -0,0 +1,162 @@
+package com.arcgismaps.toolkit.authentication
+
+import androidx.activity.ComponentActivity
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import com.arcgismaps.ArcGISEnvironment
+import com.arcgismaps.exceptions.OperationCancelledException
+import com.arcgismaps.httpcore.authentication.ArcGISAuthenticationChallenge
+import com.arcgismaps.httpcore.authentication.ArcGISAuthenticationChallengeResponse
+import com.arcgismaps.httpcore.authentication.OAuthUserConfiguration
+import com.arcgismaps.httpcore.authentication.OAuthUserCredential
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.async
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.runTest
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+/**
+ * Tests the default configuration of the [Authenticator] with OAuth.
+ *
+ * @since 200.5.0
+ */
+class OAuthDefaultConfigurationTests {
+
+    @get:Rule
+    val composeTestRule = createAndroidComposeRule<ComponentActivity>()
+
+    @Before
+    fun signOutBefore() = signOut()
+
+    @After
+    fun signOutAfter() = signOut()
+
+    private fun signOut() {
+        runBlocking {
+            ArcGISEnvironment.authenticationManager.signOut()
+        }
+        // reset the ArcGISHttpClient to remove any custom interceptors
+        ArcGISEnvironment.configureArcGISHttpClient()
+    }
+
+    /**
+     * Given an [AuthenticatorState] configured with an [OAuthUserConfiguration] for ArcGIS Online,
+     * When an [ArcGISAuthenticationChallenge] is received and the OAuth browser redirects with a successful response,
+     * Then the [ArcGISAuthenticationChallengeResponse] should be [ArcGISAuthenticationChallengeResponse.ContinueWithCredential].
+     *
+     * Note: This function automatically redirects to the app with a successful response without user input,
+     * and fakes the token validation response.
+     *
+     * @since 200.5.0
+     */
+    @Test
+    fun successfulSignIn() = runTest {
+        val response = testOAuthChallengeWithStateRestoration {
+            InstrumentationRegistry.getInstrumentation().context.startActivity(getSuccessfulRedirectIntent("kotlin-authentication-test-1://auth"))
```

Comment:
Could you add a comment here, something along the lines of...
```suggestion
            // When the OAuth sign in dialog shows up, we simulate successful sign in by launching an intent with the redirect URl, which otherwise would be sent from the Portal server, but would require valid credentials
            InstrumentationRegistry.getInstrumentation().context.startActivity(getSuccessfulRedirectIntent("kotlin-authentication-test-1://auth"))
```